File: backend/src/main/java/com/shop/service/user/UserService.java
package com.shop.service.user;

import com.shop.domain.dto.auth.AuthResponse;
import com.shop.domain.dto.user.PasswordDTO;
import com.shop.domain.dto.user.UserDTO;
import com.shop.domain.user.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface UserService {

    AuthResponse generateAuthResponse(String username, boolean isAdministration);

    Boolean createUser(UserDTO userDTO);

    User getCurrentUser();

    UserDTO getUserById( String userId);

    Page<UserDTO> getUsers(String type, Pageable pageable);

    UserDTO updateUser(String userId, UserDTO userDTO);

    List<UserDTO> getEmployees ();

    UserDTO changePassword (String userId, PasswordDTO passwordDTO);

}

File: frontend/front-admin/src/app/components/users/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { PageService } from '../page.service';
import { UtilService } from '../util.service';
import { environment } from 'src/environments/environment';
import { catchError, throwError } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient,
    private pageService: PageService,
    private utilService: UtilService) {
  }


  public getUsers(params) {
    return this.http.get(environment.apiUrl + '/user', {
      params: this.utilService.generatePageParams(params),
      observe: 'response'
    })
  }

  public createUser(formData: any) {
    return this.http.post(environment.apiUrl + '/user', formData);
  }

  public getUserById(userId: string) {
    return this.http.get(environment.apiUrl + '/user/' + userId)
  }

  public updateUser(userId: string, user: any) {
    return this.http.put(environment.apiUrl + '/user/' + userId, user)
  }

  public changePassword (userId: string, password:any){
   return this.http.patch(`${environment.apiUrl}/user/${userId}`, password).pipe(
      catchError((error) => {
        return throwError(()=> 'Error changing password');  
      }))
  }
}
